### ISSUE TITLE
Removing remote followers should be possible without blocking/unblocking

### ISSUE THREAD

**Issue by ThibG** [tootsuite/mastodon#5708]
Currently, Mastodon enables a user to remove remote followers based on domain by blocking then unblocking them.

However, such an approach has various issues (with the ActivityPub codepath):
- Block/unblocks can be sent and received out-of-order. Looking at the current code, it seems like any received “block” will be ignored if the corresponding “unblock” is processed first. This may lead the receiving instance to not actually remove local followers.
- While this is somewhat unlikely, any issue preventing the originating instance to issue an unblock request after having issued a block request will make those previous followers blocked. Same if the receiving instances experiences issues with Redis.

The OStatus codepath looks worse, as it seems like it will process block and unblocks in an arbitrary order, potentially blocking the user after “unblocking” it.

Unfortunately, it is not clear to me how that should be handled in ActivityPub. This would be needing some discussion, but I can see a few different ways this could be done:
- `Undo`-ing the `Accept`
- `Reject`-ing the `Follow` request, even though it's been accepted earlier
- `Remove` the follower from the special followers collection, but the wording in ActivityPub is confusing: the collection has to be owned by the receiving server? Even in *S2S*?

**Comment by puckipedia** [comment]
My suggestion (and the way I already implement in Kroeg) is to reject the Follow object, as this is already used to reject follow requests etc. It also keeps the amount of IDs to keep track of low (no `undo { accept { follow { user } } }`)

**Comment by ThibG** [comment]
@puckipedia right, that makes sense. Unfortunately, in Mastodon, follow requests and follow relations are two different things, and the request is destroyed as soon as it has been accepted (or rejected), so we lose track of its identifier.

**Comment by puckipedia** [comment]
haha, I have to solve that issue because the Accepts that are being sent use the wrong follow request id, and Kroeg really doesn't like that (I have added a tiny fix for this quirk, but it breaks the frontend and exact relations between objects), so I would suggest storing the follow request ID in the follow :P

**Comment by Cassolotl** [comment]
I can't find anything in the web UI for removing followers. Can anyone tell me how to do it?

**Comment by ThibG** [comment]
Unfortunately, this is not possible from Mastodon without blocking/unblocking (there is also in the settings page a way to remove followers of a given domain, but it also relies on block/unblock).
This issue was about the changes necessary for processing inbound requests to force-unfollow someone.

**Comment by Cassolotl** [comment]
Ahh okay, I will start a new issue. :) Thanks @ThibG!
